### PR TITLE
Set up Web module using Spring

### PR DESCRIPTION
**Pull request by casperboone**
This PR adds Spring in a separate web module.

The controllers for the web application will be part of relevant modules (e.g. a module for reporting to developers can define its own endpoints).

For testing I had to switch back to JUnit, since there's no nice and stable to way to test Spring apps with Spek, see issue 50 on https://github.com/spekframework/spek.

### CODE REVIEW

File: modules/web/build.gradle [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		kotlinVersion = '1.2.41'
+		springBootVersion = '2.0.2.RELEASE'
+	}
+	repositories {
+		mavenCentral()
+	}
+	dependencies {
+		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+		classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlinVersion}")
+		classpath("org.jetbrains.kotlin:kotlin-allopen:${kotlinVersion}")
```
Comment (FWDekker) [inline modules/web/build.gradle]: Why do you include Kotlin again? Isn't this already done by the root module's script?

File: modules/web/build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
+buildscript {
+	ext {
+		kotlinVersion = '1.2.41'
+		springBootVersion = '2.0.2.RELEASE'
```
Comment (FWDekker) [inline modules/web/build.gradle]: Could you move this to the `gradle.properties`?

File: modules/web/build.gradle [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${kotlinVersion}")
+		classpath("org.jetbrains.kotlin:kotlin-allopen:${kotlinVersion}")
+	}
+}
+
+apply plugin: 'kotlin'
+apply plugin: 'kotlin-spring'
+apply plugin: 'eclipse'
+apply plugin: 'org.springframework.boot'
+apply plugin: 'io.spring.dependency-management'
```
Comment (FWDekker) [inline modules/web/build.gradle]: Please use double quotes

File: modules/web/build.gradle [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+group = 'org.cafejojo.schaapi'
```
Comment (FWDekker) [inline modules/web/build.gradle]: Starting here...

File: modules/web/build.gradle [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+		freeCompilerArgs = ["-Xjsr305=strict"]
+		jvmTarget = "1.8"
+	}
+}
+compileTestKotlin {
+	kotlinOptions {
+		freeCompilerArgs = ["-Xjsr305=strict"]
+		jvmTarget = "1.8"
+	}
+}
```
Comment (FWDekker) [inline modules/web/build.gradle]: ...and ending here, the code is redundant because of the root's script.

File: modules/web/build.gradle [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+repositories {
+	mavenCentral()
+}
+
+
```
Comment (FWDekker) [inline modules/web/build.gradle]: Please remove this double line.

File: modules/web/build.gradle [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+dependencies {
+	compile('org.springframework.boot:spring-boot-starter-web')
+	compile('com.fasterxml.jackson.module:jackson-module-kotlin')
+	compile("org.jetbrains.kotlin:kotlin-stdlib-jdk8")
+	compile("org.jetbrains.kotlin:kotlin-reflect")
+	testCompile('org.springframework.boot:spring-boot-starter-test')
```
Comment (FWDekker) [inline modules/web/build.gradle]: Could you
* sort your dependencies?
* Place a newline between the `compile`s and the `testCompile`?

File: modules/web/settings.gradle [hunk 8]
```diff
@@ -0,0 +1 @@
+rootProject.name = 'web'
```
Comment (FWDekker) [inline modules/web/settings.gradle]: This file is completely redundant. Gradle doesn't even read the `settings.gradle` of subprojects.

File: modules/web/src/test/kotlin/org/cafejojo/schaapi/web/HomeTest.kt [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+class HomeTest {
+
```
Comment (FWDekker) [inline modules/web/src/test/kotlin/org/cafejojo/schaapi/web/HomeTest.kt]: Please don't add newlines at the start of classes.

File: modules/web/src/test/kotlin/org/cafejojo/schaapi/web/HomeTest.kt [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    @Autowired
+    private lateinit var restTemplate: TestRestTemplate
+
+    @Test
+    fun `it can retrieve the contents of the homepage`() {
```
Comment (FWDekker) [inline modules/web/src/test/kotlin/org/cafejojo/schaapi/web/HomeTest.kt]: That's... creative

File: modules/web/build.gradle (hunk 1 above)
Comment (casperboone) [reply modules/web/build.gradle]: I used Spring's initializer which generated this file, have not really reviewed it yet.

Will take a look at it.

**Pull request merged**